docs/topics/optimize.rst

```diff
@@ -0,0 +1,164 @@
+.. _optimize:
+
+============
+Optimization
+============
+
+Scrapy offers different ways to speed up crawls for specific :ref:`use cases
+<broad-crawls>` and to :ref:`lower resource usage <optimize-resources>`.
```

Comment:
I rewrote the broad crawl page into this page, to have a page about resource-based optimization where I would mention the `lazy` and `idle` start request scheduling strategies.

I reworked the broad crawl page content, moving aspects that seemed general about resource savings to a new section about that.

It feels a bit awkward that there is a single use-case-based fine-tuning section, but :shrug:.